packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -456,6 +464,13 @@ export class PipelineEditor extends React.Component<
     }
     const app_data = node.app_data;
 
+    if (app_data.filename !== propertySet.filename) {
+      app_data.filename = propertySet.filename;
```

Comment:
As a note: this line will need to be updated to work with #861 is merged (in whichever PR is merged last)

Reply:
Note this would specifically need to be updated to the following once #861 is merged or in #861 once this is merged.
```suggestion
      app_data.filename = PipelineService.getPipelineRelativeNodePath(
        this.widgetContext.path,
        propertySet.filename
      );
```